### DISCUSSION TITLE
CreateAsync with url parameters

### DISCUSSION THREAD

**Discussion by sivertjoe** [FirelyTeam/Vonk.Facade.Starter#65] [Q&A]
I'm trying to add a new resource to my Fhir server and I need to add a query parameter to the request, so the URL looks something like this
```
POST on {FhirBaseServer}/{resource}?_profile={SomeProfile}
```
I've tried 
```cs
await fhirClient.CreateAsync(resource, new SearchParams("_profile", profile));
```
and
```cs
var _params = new SearchParams().Add("_profile", profile);
await fhirClient.CreateAsync(resource, _params);
```
But neither seem to work, any tips for getting this to work? 😅 🙏

**Comment by mbaltus** [comment]
Could you elaborate why you think it does not work? What were you trying to achieve with the parameter? Do you get any errors from the server?

The SearchParams will be added by the FhirClient as an 'If-None-Exist' header, not as parameter on the URL. The existing resources on the server will be checked to see if there's already a resource with that value present. If not, the create will be done, if so, you can expect either a 200 OK or an error code response. This is according to the FHIR specification, see: http://hl7.org/fhir/http.html#ccreate.

**Reply by sivertjoe** [reply]
Yeah, I noticed that they were present in the If-none-exist header, but my server checks for the query parameter _profile in order to it correctly on the backend. Since no query parameter is present in the URL I get a 400 back from the server, since it needs a _profile

**Reply by mbaltus** [reply]
I would say that the server is non-compliant then, since FHIR does not specify that you can add search parameters on a POST. As a side note, it would seem strange that the server would allow only 1 resource with a particular profile value, which is the intent of a conditional create.
If the server would want to check (and file) the resource based on the profile, it has other options to do so:
- it can look for the meta.profile in the resource
- if that does not exist, it can check the resource against known profiles for that type
- if absolutely necessary that the client specifies a parameter, it could define an operation that works with the _profile parameter, so you can do an operation call instead of the create interaction

Or, from your side, instead of the conditional create, you could try a conditional update which does work with search parameters on the URL.

**Reply by sivertjoe** [reply]
Hm, okay, thanks for the help. For now I'll investigate the operation option, do you have a resource for where I can see how one can  create one?
